pkg/collector/corechecks/cluster/orchestrator/collectors/k8s_cluster.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver && orchestrator
+// +build kubeapiserver,orchestrator
+
+package collectors
+
+import (
+	"sync/atomic"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator"
+
+	"k8s.io/apimachinery/pkg/labels"
+	corev1Informers "k8s.io/client-go/informers/core/v1"
+	corev1Listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+// K8sClusterCollector is a collector for Kubernetes clusters.
+type K8sClusterCollector struct {
+	informer corev1Informers.NodeInformer
+	lister   corev1Listers.NodeLister
+	meta     *CollectorMetadata
+	proc     *processors.K8sClusterProcessor
```

Comment:
nit: I don't see a big con naming it `processor` as well for clarity. As it is not really long er than `informer` or `lister`